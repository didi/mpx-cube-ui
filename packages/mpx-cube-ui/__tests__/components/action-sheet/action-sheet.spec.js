const simulate = require('@mpxjs/miniprogram-simulate')

describe('component action-sheet unit test', function () {
  const componentId = simulate.loadMpx('src/components/action-sheet/index.mpx')
  const componentTemplateId = simulate.loadMpx('test/components/action-sheet/template/test.mpx')

  function changeProps() {
    const baseProps = {
      inputData: [
        {
          content: 'align-center',
          class: 'cube-foo'
        },
        {
          content: 'align-left',
          align: 'left'
        },
        {
          content: 'align-right',
          align: 'right'
        }
      ],
      title: '我是标题',
      cancelTxt: '取消按钮',
      active: 1
    }
    return baseProps
  }

  function newComponent(id, props) {
    const component = simulate.render(id, props)
    const parent = document.createElement('parent')
    component.attach(parent) // 会触发 attach 生命周期
    return component
  }

  describe('correct props check', () => {
    const component = newComponent(componentId, changeProps())

    it('matchSnapshot', () => {
      expect(component.dom.innerHTML).toMatchSnapshot()
    })

    it(' props to action ', async () => {
      const actionSheetItems = component.querySelectorAll('.cube-action-sheet-item')
      const itemOneAttribute = actionSheetItems[0].instance.getAttribute('data-align')
      const itemTwoAttribute = actionSheetItems[1].instance.getAttribute('data-align')
      expect(actionSheetItems.length).toBe(3)
      expect(itemOneAttribute).toBe('')
      expect(itemTwoAttribute).toBe('left')

      const itemThreeAttribute = actionSheetItems[2].instance.getAttribute('data-align')
      expect(itemThreeAttribute).toBe('right')

      const titleDom = component.querySelector('.cube-action-sheet-title').dom
      expect(titleDom.innerHTML).toBe('我是标题')

      const itemOneClassAttribute = actionSheetItems[0].dom.className
      expect(itemOneClassAttribute).toBe('main--cube-action-sheet-item main--cube-foo')

      const itemTwoClassAttribute = actionSheetItems[1].dom.className
      expect(itemTwoClassAttribute).toBe('main--cube-action-sheet-item main--cube-action-sheet-item_active')

      const cancelTxtDom = component.querySelector('.cube-action-sheet-cancel').dom
      expect(cancelTxtDom.innerHTML).toBe('取消按钮')
    })
  })

  describe('event trigger check', () => {
    const component = newComponent(componentTemplateId)

    it('matchSnapshot', () => {
      expect(component.dom.innerHTML).toMatchSnapshot() // 判断前后生成的dom是否一样
    })

    it(' wx:model ', async () => {
      // 确保 actionSheet 首次显示符合预期
      const actionSheetItems = component.querySelector('cube-action-sheet').querySelectorAll('.cube-action-sheet-item')
      expect(component.instance.active).toBe(0)
      expect(component.instance.pickerStyle).toBe(true)
      expect(actionSheetItems.length).toBe(3)
      expect(actionSheetItems[0].dom.className).toBe('cube-action-sheet--cube-action-sheet-item cube-action-sheet--cube-action-sheet-item_active')

      // 触发事件
      const itemsTwo = actionSheetItems[1]
      itemsTwo.dispatchEvent('tap')
      await simulate.sleep(10)

      expect(actionSheetItems[0].dom.className).toBe('cube-action-sheet--cube-action-sheet-item')
      expect(actionSheetItems[1].dom.className).toBe('cube-action-sheet--cube-action-sheet-item cube-action-sheet--cube-action-sheet-item_active')
      expect(component.instance.active).toBe(1)
    })
  })
})
