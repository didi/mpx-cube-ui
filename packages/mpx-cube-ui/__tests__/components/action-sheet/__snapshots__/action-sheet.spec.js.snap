// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component action-sheet unit test correct props check matchSnapshot 1`] = `"<cube-popup data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"cube-popup--cube-popup cube-popup--cube-popup_mask cube-popup--cube-popup-bottom cube-popup--cube-popup-remove-touch\\" style=\\"z-index: 100\\"><wx-view class=\\"cube-popup--cube-popup-mask\\" style=\\"\\"></wx-view><wx-view class=\\"cube-popup--cube-popup-content\\" style=\\"\\"><wx-view class=\\"main--cube-action-sheet-panel main--cube-safe-area-pb\\"><wx-view class=\\"main--cube-action-sheet-title\\" style=\\"\\">我是标题</wx-view><wx-view class=\\"main--cube-action-sheet-content\\"><wx-view class=\\"main--cube-action-sheet-list\\"><wx-rich-text data-align=\\"\\" class=\\"main--cube-action-sheet-item main--cube-foo\\" data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"rich-text\\">align-center</wx-view></wx-rich-text><wx-rich-text data-align=\\"left\\" class=\\"main--cube-action-sheet-item main--cube-action-sheet-item_active\\" data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"rich-text\\">align-left</wx-view></wx-rich-text><wx-rich-text data-align=\\"right\\" class=\\"main--cube-action-sheet-item\\" data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"rich-text\\">align-right</wx-view></wx-rich-text></wx-view></wx-view><wx-view class=\\"main--cube-action-sheet-space\\"></wx-view><wx-view class=\\"main--cube-action-sheet-cancel\\">取消按钮</wx-view></wx-view></wx-view></wx-view></cube-popup>"`;

exports[`component action-sheet unit test event trigger check matchSnapshot 1`] = `"<wx-view class=\\"main--action-sheet-demo\\"><cube-action-sheet class=\\"main--ref_actionSheet_1\\"><cube-popup data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"cube-popup--cube-popup cube-popup--cube-popup_mask cube-popup--cube-popup-bottom cube-popup--cube-popup-remove-touch\\" style=\\"z-index: 100\\"><wx-view class=\\"cube-popup--cube-popup-mask\\" style=\\"\\"></wx-view><wx-view class=\\"cube-popup--cube-popup-content\\" style=\\"\\"><wx-view class=\\"cube-action-sheet--cube-action-sheet-panel cube-action-sheet--cube-safe-area-pb\\"><wx-view class=\\"cube-action-sheet--cube-action-sheet-title cube-action-sheet--cube-action-sheet-title-picker\\" style=\\"\\">我是标题~~~</wx-view><wx-view class=\\"cube-action-sheet--cube-action-sheet-content\\"><wx-view class=\\"cube-action-sheet--cube-action-sheet-list\\"><wx-rich-text data-align=\\"\\" class=\\"cube-action-sheet--cube-action-sheet-item cube-action-sheet--cube-action-sheet-item_active\\" data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"rich-text\\">舒适型</wx-view></wx-rich-text><wx-rich-text data-align=\\"\\" class=\\"cube-action-sheet--cube-action-sheet-item\\" data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"rich-text\\">七座商务</wx-view></wx-rich-text><wx-rich-text data-align=\\"\\" class=\\"cube-action-sheet--cube-action-sheet-item\\" data-eventconfigs=\\"[object Object]\\"><wx-view class=\\"rich-text\\">豪华型</wx-view></wx-rich-text></wx-view></wx-view><wx-view class=\\"cube-action-sheet--cube-action-sheet-space cube-action-sheet--cube-action-sheet-space-picker\\"></wx-view><wx-view class=\\"cube-action-sheet--cube-action-sheet-cancel cube-action-sheet--cube-action-sheet-cancel-picker\\">取消</wx-view></wx-view></wx-view></wx-view></cube-popup></cube-action-sheet><cube-toast class=\\"main--ref_selectToast_2\\"></cube-toast></wx-view>"`;
